src/Helpers/FiltersHelper.php

```diff
@@ -0,0 +1,224 @@
+            ContentRepository::$pullFutureContent = true;
+        }
+    }
+
+    private static function processCountFilterItems($request)
+    {
+        if ($request->has('count_filter_items')) {
+            ContentRepository::$countFilterOptionItems = 1;
+        }
+    }
+
+    private static function processEnrollmentOptions($request)
+    {
+        if ($request->has('without_enrollment')) {
+            ContentRepository::$getEnrollmentContent = !$request->input('without_enrollment');
+        }
+
+        if ($request->input('only_from_my_list') == "true") {
+            $userPlaylistsService = app()->make(UserPlaylistsService::class);
+            $myList = $userPlaylistsService->getUserPlaylist(
+                user()->id,
+                'user-playlist',
+                config('railcontent.brand')
+            );
+            ContentRepository::$includedInPlaylistsIds =
+                $myList->pluck('id')
+                    ->toArray();
+        }
+    }
+
+    private static function processAdditionalFilters($request)
+    {
+        // Process 'term' filter
+        if ($request->has('term')) {
+            self::$requiredFields[] = 'name,%'.$request->input('term').'%,string,like';
+        }
+
+        // Process 'is_all' filter
+        if ($request->input('is_all') === "true") {
+            self::$requiredFields[] =
+                'published_on,'.
+                Carbon::now()
+                    ->subMonth(3)
+                    ->toDateTimeString().
+                ',date,>=';
+        }
+
+        // Process other filters based on 'tabs' input
+        if ($tabs = $request->input('tabs', $request->input('tab', false))) {
+            $tabs = is_array($tabs) ? $tabs : [$tabs];
+            foreach ($tabs as $tab) {
+                // Process each tab filter
+                self::processTabFilter($tab);
+            }
+        }
+
+        // Process 'title' filter
+        if ($request->has('title')) {
+            self::processTitleFilter($request);
+        }
+    }
+
+    /**
+     * @param $tab
+     */
+    private static function processTabFilter($tab)
+    {
+        $extra = explode(',', $tab);
+        if ($extra[0] == 'group_by') {
+            self::$groupBy = $extra[1];
+        }
+        if ($extra[0] == 'duration') {
+            self::$requiredFields[] = 'length_in_seconds,'.$extra[1].',integer,'.$extra[2].',video';
+        }
+        if ($extra[0] == 'length_in_seconds' || $extra[0] == 'topic') {
+            self::$requiredFields[] = $tab;
+        }
+        if (count($extra) == 1 && $extra[0] == 'complete') {
+            self::$requiredUserStates[] = 'completed';
+        }
+        if (count($extra) == 1 && $extra[0] == 'inProgress') {
+            self::$requiredUserStates[] = 'started';
+        }
+    }
+
+    /**
+     * @param $request
+     */
+    private static function processTitleFilter($request)
+    {
+        $title = $request->input('title');
```

Comment:
Just to be a little more consistent with other private functions, and to be more self-contained, I wonder if it might be better to pass the title in as the param, instead of the request
```php
if ($request->has('title')) {
    self::processTitleFilter($request->input('title');
}
```
```suggestion

    /**
     * @param string $title
     */
    private static function processTitleFilter(string $title)
    {
```